### DISCUSSION TITLE
i2s SAA7220 = tda1541

### DISCUSSION THREAD

**Discussion by parabyte** [pschatzmann/arduino-audio-tools#1496] [Q&A] [answered]
I am trying to use a old phillips cd player as a dac with a "esp32"
the sound quality is fantastic except for the clicking/ticking sound
its the same as the faster ticking sound like here https://github.com/pschatzmann/arduino-audio-tools/discussions/931
 i am using a level shifter it was sold as being specifically for i2c but i am under the impression its fast enough for i2s.

I cant figure out whats causing this ticking sound? is my esp32 too slow? 

I do not believe its the wifi interferring with the dac, as when i stop playback the ticking stops with the samples.

I am a newbie to digital audio so you are aware :)

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you double check the technical sheet of your sound chip to figure out how to configure it and what format it exactly needs. 

Since I never had this kind of problem, I would not know what's causing it.

Apart from this, I think you should go for this setting:  _**With OB/TWC connected to VDD the mode is the same but the data format must be in the two’s complement.**_

**Comment by parabyte** [comment]
I ditched the SAA7220 and took inspiration from what you wrote and just connected directly to the 1541. 

thank you, sound quality is exceptional!
